### DISCUSSION TITLE
Windows Compilation

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#135] [Q&A] [answered]
I try to compile BiblioteQ from linux for the windows target
with the file biblioteq.win.pro

Do you compile from the native windows system or in cross-compilation ?

if cross-compilation, some help would be welcome otherwise, I keep on looking :-)

**Comment by textbrowser** [comment]
Only for Android. Everything else is on the hardware. Even for Pi.

**Comment by textbrowser** [comment, marked as answer]
For Windows, I use a virtual machine.

**Comment by meteos77** [comment]
I try to compile from source on windows but I get 4 error messages

I left the sqlite and yaz libraries in the biblioteq directory, should I put them somewhere else?

cannot find -lsqlite3
cannot find -lyaz5
collect2.exe: error: ld returned1 exit status
[Makefile.Release:272: release/BiblioteQ.exe] Error 1

**Comment by textbrowser** [comment]
Try this.

https://github.com/textbrowser/biblioteq/blob/master/Documentation/Windows%20Installation.pdf

**Comment by meteos77** [comment]
Great: 1st compilation of my life on windows!
your documentation is OK thanks a lot.

Precision: compilation done on windows10 - qt 5.15.2
 
5 minutes of compilation for make
< 1 minutes to make install

For the last part
Remove all temporary directories from the release folder.
- the *.o files are to be deleted
- the *.cpp files are to be deleted
- the .h file to be deleted


For the following files no problem we keep :
BiblioteQ.exe
vc_redist.x64.exe
vc_redist.x86.exe
*.dll OK
biblioteq.conf
qt.conf

the file Microsoft.VC90.CRT.manifest ?

**Reply by textbrowser** [reply]
I think a mingw32-make clean should be added.

**Comment by textbrowser** [comment]
The manifest file is sometimes necessary for some people so that's why I included it.

**Comment by textbrowser** [comment]
I added a mingw32-make install step. This ought to remove the temporary files from the release folder.

**Comment by meteos77** [comment]
I don't have the poppler library, or should I get it ?


the -j option of the compiler works
win make -j4 -> 90 seconds (virtualbox) 
linux make -j4 ->  53 seconds

**Comment by textbrowser** [comment]
Very difficult to get all of its dependencies on Windows. Poppler.

**Comment by textbrowser** [comment]
Poppler.

https://repo.msys2.org/mingw/i686/

I'll see if I can integrate it again.

**Comment by textbrowser** [comment]
![Screenshot_20221130_103752](https://user-images.githubusercontent.com/10701156/204842611-92476977-5d12-4e62-a8a9-2d3cd083ac3c.png)

**Comment by meteos77** [comment]
can you explain me what exactly poppler does because we can compile without the library?
what functionality is impacted by this absence ?

**Comment by textbrowser** [comment]
Poppler is a PDF-rendering library. It's optional. All dependencies in BQ are now optional, except for Qt. Poppler, PostgreSQL, YAZ are all optional.

With Poppler, BQ allows viewing and searching of PDF files. Qt has a PDF library which is sometimes available. I integrated it into Glitch for Linux and Mac. One can view PDF files but not search them. Well, there are better applications for that anyway.

**Comment by meteos77** [comment]
When compiling, mingw32-make install
the process creates the directory biblioteq/release/plugins/ + 28 subdirectories and inside the .dll.debug files

how to avoid them because I think they are useless in my case?

**Comment by textbrowser** [comment]
I don't see these files.

**Comment by textbrowser** [comment]
pluginspurge.extra = del /Q /S *d.dll
pluginspurge.path = release\\plugins\\.

In project file.

**Reply by meteos77** [reply]
pluginspurge.extra = del /Q /S *d.dll
pluginspurge.path = release\\plugins\\.

Do the two backslash make a difference?

**Comment by textbrowser** [comment]
Rebuilt now. I don't have debug libraries.

**Reply by meteos77** [reply]
it is at the time of the make install that it creates the plugins directory
![1](https://user-images.githubusercontent.com/56431039/205993622-d6fd6b9b-c470-443e-9fed-118e7c5aed23.png)

**Comment by meteos77** [comment]
![3](https://user-images.githubusercontent.com/56431039/205993887-b1a7812a-4631-4a99-aaba-39e9ad564b83.png)

**Reply by meteos77** [reply]
65Mo

**Comment by textbrowser** [comment]
I don't have dll.debug libraries in my Qt installation. If I don't have them, I cannot delete them.

**Comment by textbrowser** [comment]
I don't know how you installed Qt. I don't know which modules you selected. This information is hidden to me. I can only purge files which are included in my very basic Qt installation.

**Reply by meteos77** [reply]
I will add this in my doc, if the files are useless
`del /s release\plugins\*.debug`

**Reply by textbrowser** [reply]
Or fork and provide a revision of biblioteq.win.pro. :P

**Reply by meteos77** [reply]
I compile on windows10 since about 1 month, and it's to be up to date with BiblioteQ with the library's pc and mine.
I can thus show the last evolutions of BQ without waiting for your windows package.

if it was up to me, pc library = linux :-)

**Reply by meteos77** [reply]
bye bye windows

**Reply by textbrowser** [reply]
In 2023, my releases will be source only for all programs.

**Reply by textbrowser** [reply]
Creating a release that requires 8 hours of my life is stoooopid. Done. Here are the files and enjoy.

**Reply by meteos77** [reply]
maybe there will be motivated people to take over from you. 

I dream of becoming an apt install biblioteq :-)

**Reply by textbrowser** [reply]
I'm not going to hold by breath. People want free and I'm done with that.

**Reply by meteos77** [reply]
Is BiblioteQ going to become a paid service?

**Reply by textbrowser** [reply]
Of course not. People will have to build it on their own, package it on their own, etc. If not, find free or paid support or move on towards other non-free projects. PowerPC, Android, numerous Linuxes, Pi, Windows. I won't prefer one over the other.

**Reply by meteos77** [reply]
ah ok, you are not interested in distributing the software because it is too time consuming

**Reply by textbrowser** [reply]
It is. It takes many hours of building, packaging, uploading.

**Reply by textbrowser** [reply]
People will lose interest because it's not available on their computer, but that's the nature. Books are not always available in your language. Caring contributors are required.

**Reply by meteos77** [reply]
there are already the 2 discussions on the compilation
Windows #135
LinuxMint #239

I would make a new compilation on raspberry taking notes to get the right package names

**Reply by meteos77** [reply]
people who compile on android with their tablet?

**Reply by textbrowser** [reply]
Beyond family. I want to return to my piano, want to read more, more gaming, more puzzling. Packaging things is not a hobby. Consider a farmer growing food, harvesting it, packaging it, and then distributing it for free to people. Insane.

**Reply by textbrowser** [reply]
> people who compile on android with their tablet?

They need to learn, fail, experiment, and succeed. Frustration is fun.

**Reply by textbrowser** [reply]
And anyway, I don't have to justify myself! Suppose someone's giving away a free television and they publish that it's used and somewhat damaged and the person receiving it complains. Insane.

**Reply by meteos77** [reply]
You still need some knowledge of computers to compile a program. But the lack of package will attract volunteers.

I find the forum not animated enough except for you and me :-)

We'll see what happens next.

**Reply by textbrowser** [reply]
Riiiigggggggggggggggghttttttttttttt. Project may lose interest or interest may increase because people are not handed everything.

**Reply by meteos77** [reply]
You have no idea how many people download the packages?

I do if I use BiblioteQ for my library and great new features appear. I'll be interested in compiling to get them soon enough :-)

**Reply by textbrowser** [reply]
I don't and I don't review metrics. Let me see if I can find that information.

**Reply by meteos77** [reply]
![1site-de-telechargement](https://user-images.githubusercontent.com/56431039/206004093-f16b8452-c8ee-433d-9afe-f3158901f963.png)
on a download site : 11963

**Reply by textbrowser** [reply]
Someone informed me.

https://tooomm.github.io/github-release-stats/?username=textbrowser&repository=biblioteq

**Reply by textbrowser** [reply]
Softpedia is shady.

**Reply by meteos77** [reply]
541 at the source on github !
what surprises me, the number of active participants in the forum?

**Reply by textbrowser** [reply]
I mean I like providing files. However! I cannot automate it because my machines are always off. Well, I could. Say, by support daemons which activate something if a file is present. For Windows? Yuck. Then there are the uploads, release notes, tagging. Yuck.

**Reply by textbrowser** [reply]
I would prefer silence because that would suggest that it meets everyone's needs or that there is zero interest. Win-win!

**Reply by meteos77** [reply]
it is the case has by a French user, who has unusual needs?

**Reply by textbrowser** [reply]
Maybe you have above-average attention.

**Reply by meteos77** [reply]
I hope that the user community will put something in place, to support you in making packages, which will allow newcomers to enjoy the great software you create for us.

Thanks again.
For me the day ends ... good continuation

**Reply by textbrowser** [reply]
Do you live in the library?

**Reply by textbrowser** [reply]
I would enjoy living above a library. Cozy.

**Comment by textbrowser** [comment]
https://doc.qt.io/qt-5/windows-deployment.html

**Comment by textbrowser** [comment]
Committed.
